src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1315,20 +1315,25 @@ CHIP_ERROR CASESession::GetHardcodedTime()
 
 CHIP_ERROR CASESession::SetEffectiveTime()
 {
-    System::Clock::Milliseconds64 currentTimeMS;
-    CHIP_ERROR err = System::SystemClock().GetClock_RealTimeMS(currentTimeMS);
-    if (err == CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE)
+    System::Clock::Milliseconds64 timeMS;
+    CHIP_ERROR err = System::SystemClock().GetClock_RealTimeMS(timeMS);
```

Comment:
It looks like this code expects `System::SystemClock().GetClock_RealTimeMS` to be in Unix-epoch time. This is problematic since unix epoch is not part of the spec at all. I am surprised we would use our platform abstraction APIs use Unix time when many targts run on embedded class devices with no actual legacy needs of Unix epoch as a basis.

Overall, these conversions are also error-prone and not significantly tested.